src/components/Weather.js

```diff
@@ -0,0 +1,15 @@
+import React from "react";
+//redux
+import { useSelector } from "react-redux";
+
+const Weather = ({ children }) => {
+  const { isLoading } = useSelector((state) => state.weather);
+
+  return (
+    <div>
+      <>{!isLoading && <div>{children}</div>}</>
```

Comment:
Why use the parent/child here? Couldn't everything from `WeatherData` just go here?